contents/stats.md

```diff
@@ -0,0 +1,445 @@
+Intuitively, the median tells us the value of the data's 50\% percentile.
+One example is when we are analyzing income.
+The median is the upper limit that we expect that half of the observations earn.
+So, if the median income is U$ 80,000, we expect that half of our observations earn between the minimum value and U$ 80,000.
+The mathematical formula for the median is:
+
+$$ \operatorname{median}(\mathbf{x}) = \frac{x_{\lfloor (\# x+1) \div 2 \rfloor} + x_{\lceil (\# x+1) \div 2 \rceil}}{2}, $$ {#eq:median}
+
+where:
+
+- $\mathbf{x} = x_1, \cdots, x_n$ is an ordered list of numbers
+- $\# x$ is the list length
+- $\lfloor . \rfloor$ a rounded-down value to the nearest integer
+- $\lceil . \rceil$ a rounded-up value to the nearest integer
+
+Similarly, we can use the `median` from `Statistics` module to apply the median to our data:
+
+```julia
+using Statistics: median
+```
+
+```jl
+# median # hide
+s = """
+    gdf = groupby(more_grades(), :name)
+    combine(gdf, :grade => median)
+    """
+sco(s; process=without_caption_label)
+```
+
+As we can see, the median differs substantially from the mean.
+
+### Mode {#sec:stats_central_mode}
+
+The mean and median can be useful for numerical and ordinal data.
+However, they are ineffective for nominal data, in which our data is comprised of qualitative data (also known as categorical data).
+This is where we use the **mode**, defined as **the most frequent value in our data**.
+
+For the mode, we *do not* have a `mode` function inside Julia's standard library `Statistics` module.
+Instead, we need to use the `StatsBase.jl` for less common statistical functions:
+
+```julia
+using StatsBase: mode
+```
+
+In @sec:missing_data, we have the `correct_types` `DataFrame`, which is mainly categorical with `Date`s and `String`s:
+
+```jl
+# inside stats content # hide
+sco("correct_types()"; process=without_caption_label)
+```
+
+We can compute the mode with the `combine` function from `DataFrames.jl`.
+Notice that, unlike previously with mean and median, we will not group our data:
+
+```jl
+# mode # hide
+s = """
+    combine(correct_types(), :age => mode)
+    """
+sco(s; process=without_caption_label)
+```
+
+### Visualization of Central Tendencies {#sec:stats_central_vis}
+
+In order to have a better intuition behind the difference between mean, median and mode, it is always useful to make some visualizations.
+We will cover statistical visualizations in depth in @sec:stats_vis.
+Below, in @fig:plot_central, we have two data distributions:
+
+- **Upper row**: **normal** distributed data
+- **Lower row**: **non-normal** distributed data
+
+```jl
+s = """
+    fig = plot_central()
+    caption = "Normal and Non-Normal Distributed Data -- Differences Between Central Tendencies."
+    label = "plot_central"
+    Options(fig; filename=label, caption, label)
+    """
+sco(s)
```

Comment:
```suggestion
fig = plot_central()
caption = "Normal and Non-Normal Distributed Data -- Differences Between Central Tendencies."
label = "plot_central"
Options(fig; filename=label, caption, label)
```
This hides the code for the plot from the reader. If you want to show the code for `plot_central`, use `@sco plot_central()` and inside that function:
```
function plot_central()
     fig = ...
     return Options(fig; filename=label, caption, label) # hide
end
```